File: src/main/java/com/aladdinworks5/service/TemperatureAlertService.java
package com.aladdinworks5.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.aladdinworks5.domain.TemperatureAlert;
import com.aladdinworks5.dto.TemperatureAlertDTO;
import com.aladdinworks5.dto.TemperatureAlertSearchDTO;
import com.aladdinworks5.dto.TemperatureAlertPageDTO;
import com.aladdinworks5.dto.TemperatureAlertConvertCriteriaDTO;
import com.aladdinworks5.service.GenericService;
import com.aladdinworks5.dto.common.RequestDTO;
import com.aladdinworks5.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface TemperatureAlertService extends GenericService<TemperatureAlert, Integer> {

	List<TemperatureAlert> findAll();

	ResultDTO addTemperatureAlert(TemperatureAlertDTO temperatureAlertDTO, RequestDTO requestDTO);

	ResultDTO updateTemperatureAlert(TemperatureAlertDTO temperatureAlertDTO, RequestDTO requestDTO);

    Page<TemperatureAlert> getAllTemperatureAlerts(Pageable pageable);

    Page<TemperatureAlert> getAllTemperatureAlerts(Specification<TemperatureAlert> spec, Pageable pageable);

	ResponseEntity<TemperatureAlertPageDTO> getTemperatureAlerts(TemperatureAlertSearchDTO temperatureAlertSearchDTO);
	
	List<TemperatureAlertDTO> convertTemperatureAlertsToTemperatureAlertDTOs(List<TemperatureAlert> temperatureAlerts, TemperatureAlertConvertCriteriaDTO convertCriteria);

	TemperatureAlertDTO getTemperatureAlertDTOById(Integer temperatureAlertId);







}

File: aladdinworks5_ui/src/services/TemperatureAlertService.js
import http from "../http-common"; 

class TemperatureAlertService {
  getAllTemperatureAlerts(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/temperatureAlert/temperatureAlerts`, searchDTO);
  }

  get(temperatureAlertId) {
    return this.getRequest(`/temperatureAlert/${temperatureAlertId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/temperatureAlert?field=${matchData}`, null);
  }

  addTemperatureAlert(data) {
    return http.post("/temperatureAlert/addTemperatureAlert", data);
  }

  update(data) {
  	return http.post("/temperatureAlert/updateTemperatureAlert", data);
  }
  
  uploadImage(data,temperatureAlertId) {
  	return http.postForm("/temperatureAlert/uploadImage/"+temperatureAlertId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new TemperatureAlertService();
